tests/unit/test_charm.py

```diff
@@ -516,3 +569,103 @@ def test_on_client_config_changed_with_external_url_config(harness, mocked_conta
             "redirect_uri": f'{harness.charm.config["external_url"]}/self-service/methods/oidc/callback/{provider_id}',
         }
     ]
+
+
+def test_on_client_config_changed_with_hydra(harness) -> None:
+    setup_postgres_relation(harness)
+
+    container = harness.model.unit.get_container(CONTAINER_NAME)
+    harness.charm.on.kratos_pebble_ready.emit(container)
+
+    setup_hydra_relation(harness)
+
+    expected_config = {
+        "log": {"level": "trace"},
+        "identity": {
+            "default_schema_id": "default",
+            "schemas": [
+                {"id": "default", "url": "file:///etc/config/identity.default.schema.json"}
+            ],
+        },
+        "selfservice": {
+            "default_browser_return_url": "http://127.0.0.1:4455/",
+            "flows": {
+                "error": {
+                    "ui_url": "http://127.0.0.1:4455/oidc_error",
+                },
+                "login": {
+                    "ui_url": "http://127.0.0.1:4455/login",
+                },
+                "registration": {
+                    "enabled": True,
+                    "ui_url": "http://127.0.0.1:4455/registration",
+                },
+            },
+        },
+        "dsn": f"postgres://{DB_USERNAME}:{DB_PASSWORD}@{DB_ENDPOINTS}/{harness.model.name}_{harness.charm.app.name}",
+        "courier": {
+            "smtp": {"connection_uri": "smtps://test:test@mailslurper:1025/?skip_ssl_verify=true"}
+        },
+        "serve": {
+            "public": {
+                "base_url": "None",
+                "cors": {
+                    "enabled": True,
+                },
+            },
+        },
+        "oauth2_provider": {
+            "url": "http://hydra-admin-url:80/testing-hydra",
+        },
+    }
+
+    assert yaml.safe_load(harness.charm._render_conf_file()) == expected_config
+
+
+def test_on_client_config_changed_when_missing_hydra_relation_data(harness) -> None:
```

Comment:
Pretty close to a `given_x_when_y_then_z`!! I'd definitely encourage going that route :) Here it'd look like

`given_missing_hydra_relation_data_when_config_changed_then_config_file_is_rendered`

Reply:
I was trying to follow the existing naming convention, should I rename all tests?

Reply:
Ultimately I think this naming convention is a good one (`given_x_when_y_then_z`) but it's not as if it was widely used at Canonical so I don't wan't to force it on you.